cli/dcoscli/config/main.py

```diff
@@ -237,19 +252,31 @@ def _unset(name, index):
         if index is not None:
             index = util.parse_int(index)
 
+            if not value:
+                raise DCOSException(
+                    'Index ({}) is out of bounds - {} is empty'.format(
+                        index,
+                        name))
             if index < 0 or index >= len(value):
                 raise DCOSException(
                     'Index ({}) is out of bounds - possible values are '
                     'between {} and {}'.format(index, 0, len(value) - 1))
 
-            value.pop(index)
+            popped_value = value.pop(index)
+            emitter.publish(
+                "[{}]: removed element '{}' at index '{}'".format(
```

Comment:
We might want to add another case lists of length 1 - this output is a little odd:
```➜  cli git:(DCOS-2379-show-config-changes) ✗ dcos config unset --index=2 package.sources
Index (2) is out of bounds - possible values are between 0 and 0```